### ISSUE TITLE
Certain XCompose values fail

### ISSUE THREAD

**Issue by willismonroe** [ibus/ibus#1990]
Please fill in the following items if you don't know the root cause.

Which distribution and version?: 
Ubuntu 17.10

Which desktop environment and version?: 
Gnome 3.26

Which session type?: 
x11

Which application and version?: 
GNOME Terminal 3.24.2 Using VTE version 0.48.4 +GNUTLS -PCRE2

IBus version?: 
IBus 1.5.14

Issue description:
Certain lines in my XCompose file will work and others won't and I'm not sure of the root cause.

Steps to reproduce:
1. press compose key + compose key + l + e + o
2. produces the character "♌"
3. press compose key + compose key + v + i + r + g + o
4. produces the string "irgo"

Can you reproduce your problem when you restart ibus-daemon? (yes / no): 
no

Do you see any errors when you run ibus-daemon with the verbose option?:
no

Can you reproduce your problem with a new user account instead of the current your account? (yes / no):  Haven't tried. 

Here's what the relevant part of my .XCompose looks like:

```
<Multi_key> <Multi_key> <l> <e> <o>             : "♌"  U264C   # LEO
<Multi_key> <Multi_key> <v> <i> <r> <g> <o>     : "♍"  U264D   # VIRGO
```

**Comment by fujiwarat** [comment]
I cannot reproduce your problem.
When I tested the line of "♍", I commented out the line of "♌" because both line is the same shortcut key.

Can you show the value of `source` in `org.gnome.desktop.input-sources`?
```
% gsettings get org.gnome.desktop.input-sources sources
[('xkb', 'jp'), ('xkb', 'us'), ('ibus', 'anthy')]
```

If you don't configure ibus, you need gtk 3.22 or later.

How do you configure a key to the `Multi_key`?

Can you see the string of `Multi_key` when you type `Multi_key` on the [window.py.txt](https://github.com/ibus/ibus/files/1829333/window.py.txt) ?
```
% mv window.py.txt window.py
% env GTK_IM_MODULE=gtk-im-context-none python window.py
```

**Comment by willismonroe** [comment]
I updated my comment to show what I actually had in my .XCompose... I didn't realize that github would eat the single letters with carets.

```
➜  ~ gsettings get org.gnome.desktop.input-sources sources
[('xkb', 'us')]
```
Your program outputs: "Multi_key 42 10"

**Comment by fujiwarat** [comment]
> I updated my comment to show what I actually had in my .XCompose... I didn't realize that github would eat the single letters with carets.

OK.

If you remove the cache file and run GTK application again, you will see the warning about the line of "♍".

```
% rm $HOME/.cache/gtk-3.0/compose/*.cache
% python window.py
(window.py:13597): Gtk-WARNING **: The max number of sequences is 7: <Multi_key> <Multi_key> <v> <i> <r> <g> <o>     : "♍"  U264D   # VIRGO
```

It's the current limitation.
You can put less than 7 sequences. E.g.
```
<Multi_key> <v> <i> <r> <g> <o>     : "♍"  U264D   # VIRGO
```

**Comment by willismonroe** [comment]
Yeah I noticed that error, and found the constant defined by
GTK_MAX_COMPOSE_LEN here:
https://people.gnome.org/~ryanl/gtk/html/GtkIMContextSimple.html

I guess that's a major limiting factor, any reason the GTK folks set it at
7?  Should I bring this up with them... it would be nice to be able to
script longer compose sequences.

**Comment by fujiwarat** [comment]
My understanding is, the reason would be to match the typed sequences to the registered sequences more quickly.
The main algorithm is https://gitlab.gnome.org/GNOME/gtk/blob/master/gtk/gtkimcontextsimpleseqs.h for the US compose table.
I added that warning message in both GTK and IBus because IBus has integrated the same algorithm.
Probably we can enhance that code to check GTK_MAX_COMPOSE_LEN and IBUS_MAX_COMPOSE_LEN for only the user compose table but not the pre-registered US compose table.

**Comment by m93a** [comment]
I've spent way too much time [looking for the cause](https://askubuntu.com/q/1037597) of my sequences not working... (I also found out that it's not possible to output multiple characters using the Compose key. #1819)

I think that six symbols as the maximum sequence length is quite limiting. For example an intuitive way of writing integrals would be something like this:

```XCompose
<Multi_key> <Multi_key> <I> <n> <t>             : "∫"
<Multi_key> <Multi_key> <I> <I> <n> <t>         : "∬"
<Multi_key> <Multi_key> <I> <I> <I> <n> <t>     : "∭"
<Multi_key> <Multi_key> <o> <I> <n> <t>         : "∮"
<Multi_key> <Multi_key> <o> <I> <I> <n> <t>     : "∯"
<Multi_key> <Multi_key> <o> <I> <I> <I> <n> <t> : "∰"
```

One could get around this by using only one `<Multi_key>` (risking collisions with other sequences) and replacing `<o> <I> <I>` for surface integral with an S (and V for the volume integral) but it would be a lot less obvious and the user would have to search for the sequences and remember them.

Furthermore, I think that the beforementioned Virgo sequence is a perfectly valid example that should be possible to use.

I'm pretty sure that allowing more characters can be done without sacrificing performance. For example [WinCompose](https://github.com/samhocevar/wincompose) contains sequences like `♦♦snowboarder` and I've never experienced problems with it.
